src/main/scala/scorex/core/network/message/BasicMessagesRepo.scala

```diff
@@ -156,21 +158,22 @@ class ModifiersSpec(maxMessageSize: Int) extends MessageSpecV1[ModifiersData] wi
       w.putBytes(modifier)
     }
 
-    if (msgSize > maxMessageSize) {
-      log.warn(s"Message with modifiers ${modifiers.keySet} has size $msgSize exceeding limit $maxMessageSize.")
+    if (msgSize > maxMsgSizeWithReserve) {
+      log.warn(s"Message with modifiers ${modifiers.keySet} has size $msgSize exceeding limit $maxMsgSizeWithReserve.")
     }
   }
 
   override def parse(r: Reader): ModifiersData = {
     val typeId = ModifierTypeId @@ r.getByte() // 1 byte
     val count = r.getUInt().toIntExact // 8 bytes
+    require(count > 0, s"Illegal message with 0 modifiers of type $typeId")
```

Comment:
The same goes here, we get the Bytes but as `count` was 0 due to exceeding the maxSize, we ignore it ... so it is better to Fail Fast imo.